scripts/runLinter.lean

```diff
@@ -17,6 +18,27 @@ def readJsonFile (α) [FromJson α] (path : System.FilePath) : IO α := do
 def writeJsonFile [ToJson α] (path : System.FilePath) (a : α) : IO Unit :=
   IO.FS.writeFile path <| toJson a |>.pretty
 
+open Lake
+
+/-- Get either the first lean_exe or lean_lib which is a default Lake target-/
+def getFirstDefaultExeRootOrLib?: IO (Option Name) := do
```

Comment:
I think you are right.

I asked [a related question](https://github.com/leanprover-community/batteries/issues/795#issuecomment-2132234487) on the corresponding #795 issue asking about what criteria we should use to determine a set of modules from the default targets.

Since some default targets may not have a corresponding Lean module, it seems to me that we need to treat each build target type differently. One path forward would be to use the `root` module of any `lean_exe` default target and all modules in the `roots` array for any `lean_lib` default target.

This is my first time working directly with the Lake API, so I may be overcomplicating this problem.

Reply:
I updated the proposed changes to lint all root modules of any `lean_exe` or `lean_lib` default targets.

Note, there is a bug when a `lean_exe` default target doesn't specify a root module. I need to investigate this further.

Reply:
Okay, I have fixed this specific bug.

Additionally, I updated the proposed changes to use the `resolveDefaultPackageTarget ` function to align the design with `lake build`.